packages/tooltip/src/useTooltip.ts

```diff
@@ -44,22 +45,38 @@ export const fromEntries = (entries: any) =>
 export function useTooltip({
   placement = 'top',
   offset = [0, 10],
+  closeTimeout = 0,
 }: UseTooltipProps = {}): UseTooltipReturn {
   const triggerRef = React.useRef<HTMLElement | null>(null)
   const tooltipRef = React.useRef<HTMLElement | null>(null)
   const arrowRef = React.useRef<HTMLElement | null>(null)
   const popperInstanceRef = React.useRef<any>(null)
+  const timeoutIdRef = React.useRef<any>(null)
 
   const id = useId()
 
   const {
     isOpen: isVisible,
     handleOpenPortal,
-    handleClosePortal,
+    handleClosePortal: instantlyClosePortal,
     Portal,
     portalRef,
   } = usePortal()
 
+  const delayedClosePortal = React.useCallback(
+    function delayedClosePortal(event) {
+      clearTimeout(timeoutIdRef.current)
```

Comment:
@seri this is really minor but I was just re-reading https://overreacted.io/making-setinterval-declarative-with-react-hooks/ and it reminded me of a small edge case to prevent getting an extra react dev error "Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.".  This edge case happens if the popup is open, and we unmount the component (like hovering over something then changing tabs).

When the hook is dismounted we should also cleanup the timeout.  So to do this we'd need a 

```
React.useEffect(function cleanupTimeOut {
 return () => {
    clearTimeout(timeoutIdRef.current)
 }
})
```